src/runtime/pmix_finalize.c

```diff
@@ -148,8 +148,8 @@ void pmix_rte_finalize(void)
     PMIX_LIST_DESTRUCT(&pmix_globals.nspaces);
     PMIX_LIST_DESTRUCT(&pmix_client_globals.groups);
 
-    for (i=0; i < pmix_globals.keyindex.size; i++) {
-        p = (pmix_regattr_input_t*)pmix_pointer_array_get_item(&pmix_globals.keyindex, i);
+    for (i=0; i < pmix_globals.keyindex.table->size; i++) {
+        p = (pmix_regattr_input_t*)pmix_pointer_array_get_item(pmix_globals.keyindex.table, i);
```

Comment:
Might as well just put this in the pmix_keyindex_t destructor?

Reply:
Makes perfect sense! I've included change this in my latest push.